File: eca-server/src/main/java/com/ecaservice/server/service/ClassifyInstanceService.java
package com.ecaservice.server.service;

import com.ecaservice.web.dto.model.ClassifyInstanceRequestDto;
import com.ecaservice.web.dto.model.ClassifyInstanceResultDto;

/**
 * Classify instance service.
 *
 * @author Roman Batygin
 */
public interface ClassifyInstanceService {

    /**
     * Classify specified instance.
     *
     * @param classifyInstanceRequestDto - classify instance request dto
     * @return classify instance result
     */
    ClassifyInstanceResultDto classifyInstance(ClassifyInstanceRequestDto classifyInstanceRequestDto);
}

File: eca-web/src/main/resources/src/app/common/services/classify-instance.service.ts
import {
  ClassifyInstanceRequestDto, ClassifyInstanceResultDto
} from "../../../../../../../target/generated-sources/typescript/eca-web-dto";
import { Observable } from "rxjs/internal/Observable";

export interface ClassifyInstanceService {

  classifyInstance(classifyInstanceRequestDto: ClassifyInstanceRequestDto): Observable<ClassifyInstanceResultDto>;
}
